modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/common/services/SpaceService.ts

```diff
@@ -27,18 +27,37 @@ async function addSpace({
 	);
 }
 
-async function getSpace({
-	externalReferenceCode,
-	spaceId,
-}: {
+interface GetSpaceEndpointArgs {
 	externalReferenceCode?: string;
 	spaceId?: number | string;
-}): Promise<Space> {
-	let url = `/o/headless-asset-library/v1.0/asset-libraries/by-external-reference-code/${externalReferenceCode}`;
+}
+
+function getSpaceEndpoint({
+	externalReferenceCode,
+	spaceId,
+}: GetSpaceEndpointArgs) {
+	let url: string;
 
 	if (spaceId) {
 		url = `/o/headless-asset-library/v1.0/asset-libraries/${spaceId}`;
 	}
+	else if (externalReferenceCode) {
+		url = `/o/headless-asset-library/v1.0/asset-libraries/by-external-reference-code/${externalReferenceCode}`;
+	}
+	else {
+		throw new Error(
+			'Either spaceId or externalReferenceCode must be provided.'
+		);
+	}
+
+	return url;
```

Comment:
Why not use a function overload here? It would simplify the code quite a bit, and errors would be detected at compile time.
```
function getAPIEndpoint(spaceId: number);
function getAPIEndpoint(externalReferenceCode: string);
function getAPIEndpoint(id: string | number) {
    return typeof id === "string" ? `...` : `...`;
}
```